src/Tokenizers/PHP.php

```diff
@@ -893,6 +893,62 @@ protected function tokenize($string)
                 continue;
             }//end if
 
+            /*
+                Tokenize the parameter labels for PHP 8.0 named parameters as a special T_PARAM_NAME
+                token and ensure that the colon after it is always T_COLON.
+            */
+
+            if ($tokenIsArray === true
+                && preg_match('`^[a-zA-Z_\x80-\xff]`', $token[1]) === 1
```

Comment:
I'm a bit worried about running a preg_match on so many tokens, and then the following loop to find the next non-empty looking for a colon.

I probably would have tried tackling this problem starting with the colon and looking backwards, although I know that's going to be complex due to the different ways it can be used. I possibly would have also tried in processAdditional, but that might have been even more complex to unravel.

I'm curious to know if you tried any alternative implementations before I go and play around with the code to see how it's working.

Reply:
> I'm a bit worried about running a preg_match on so many token

Well, as regexes go, this is as fast you can get: a regex checking for one character on a string of one character, so no chance of backtracing at all, it either matches or it doesn't.
The alternative would be to have a list of the characters we want to exclude and do either a `strpos()` or `isset()` or something, but then that would break straight away if a new special character would get meaning in PHP. It would definitely be less stable.

> I probably would have tried tackling this problem starting with the colon and looking backward

That was my first choice for token to search for, but then the "final token" for the _previous_ effective token (which is the one we need to chance) may already have been set in a previous loop, and what with whitespace and comments allowed everywhere, that seemed like it was going to be pretty complicated.

> I possibly would have also tried in processAdditional, but that might have been even more complex to unravel.

I consider that as well, but that would have been too late to prevent misidentified ternary "else" tokens and trying to recreate the whole if/else determination in `processAdditional()` after the fact, seemed like it would cause a huge overhead, aside from it probably not being stable.

Reply:
Just thinking - I could possibly make the regex condition a "T_STRING _or_ matching the regex" condition, that way the most common token for identifiers would not be run through the regex, which would possibly be slightly faster.

Reply:
And yes, I thought long and hard before deciding on this way to do it, considered sniffing based on the `(` or `,` before it too. Every time I tried, there was something which would make it less stable or more complex to set the final token, which is why I ended up with this.

Reply:
I assumed you'd thought through those options, but just wanted to check before I dug into it. Bypassing the regex for T_STRING tokens is a good idea - I might give it a go. The *extensive* unit tests make this very easy to play around with, so thank you so much for coming up with all those test cases.